admin_ui/src/components/InputField.vue

```diff
@@ -1,5 +1,20 @@
 <template>
     <div>
+        <template v-if="isMediaColumn && !isFilter">
+            <div class="media_block">
+                <input type="file" @change="uploadFile($event)" />
+                <a href="#" @click.prevent="showMedia" v-if="localValue"
+                    >View</a
+                >
```

Comment:
I don't know if this helps, but you can write this part like this  
```vue
<a
    href="#"
    @click.prevent="showMedia"
    v-if="localValue && type !== 'array'"
    ><font-awesome-icon icon="eye" size="lg" title="View" // nedd to add faEye to fontawesome.ts
/></a>
```
so that the `View` button is not displayed in the array widget.

Reply:
This is a full template with an upload button that replaces the input file type and a view icon to display the file. I think that looks ok.
```vue
<template v-if="isMediaColumn && !isFilter">
    <div class="media_block">
        <form
            id="uploadForm"
            enctype="multipart/form-data"
            v-on:change="uploadFile($event)"
        >
            <label>
                Upload file
                <input type="file" name="file" />
            </label>
        </form>
    </div>

    <a
        href="#"
        @click.prevent="showMedia"
        v-if="localValue && type !== 'array'"
        ><font-awesome-icon icon="eye" size="lg" title="View"
    /></a>

    <MediaViewer
        v-if="showMediaViewer"
        :mediaViewerConfig="mediaViewerConfig"
        @close="showMediaViewer = false"
    />
</template>
...
<style>
div.media_block {
    display: inline-block;
}

input[type="file"] {
    display: none;
}

label {
    cursor: pointer;
    border: none;
    padding: 0.8rem 1.2rem;
    font-weight: bolder;
    margin-bottom: 0.5rem;
    transition: background-color 0.5s;
    background-color: @dark_blue;
    color: white;
    font-size: 0.7em;
    text-transform: uppercase;
    text-align: center;
}

a {
    float: right;
    margin-top: 0.7rem;
}
</style>
```
Result is
![upload_button](https://user-images.githubusercontent.com/30960668/182788199-85e7991e-c86e-4efc-8a99-765ddcbbe2c4.png)